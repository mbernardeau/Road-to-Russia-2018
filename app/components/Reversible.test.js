import React from 'react'
import renderer from 'react-test-renderer'

import Reversible from './Reversible'

describe('components/Flag', () => {
  it('renders correctly when no direction specified', () => {
    const tree = renderer
      .create(
        <Reversible>
          <div>1</div>
          <div>2</div>
        </Reversible>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with ltr', () => {
    const tree = renderer
      .create(
        <Reversible direction="ltr">
          <div>1</div>
          <div>2</div>
        </Reversible>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with rtl', () => {
    const tree = renderer
      .create(
        <Reversible direction="rtl">
          <div>1</div>
          <div>2</div>
        </Reversible>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('passes props to the underlying div', () => {
    const tree = renderer
      .create(
        <Reversible style={{ margin: 1 }}>
          <div>1</div>
          <div>2</div>
        </Reversible>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
