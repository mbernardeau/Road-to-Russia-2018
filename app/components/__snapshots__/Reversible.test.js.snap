// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Flag passes props to the underlying div 1`] = `
<div
  style={
    Object {
      "margin": 1,
    }
  }
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`components/Flag renders correctly when no direction specified 1`] = `
<div>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`components/Flag renders correctly with ltr 1`] = `
<div>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;

exports[`components/Flag renders correctly with rtl 1`] = `
<div>
  <div>
    2
  </div>
  <div>
    1
  </div>
</div>
`;
